app/api/domains/cho.py

```diff
@@ -1217,9 +1224,16 @@ async def handle(self, player: Player) -> None:
                             # use player mode if not specified
                             mode_vn = player.status.mode.as_vanilla
 
+                        # parse the mods from regex
+                        if r_match["mods"] is not None:
+                            mods = Mods.from_np(r_match["mods"][1:], mode_vn)
+                        else:
+                            mods = None
+
                         player.last_np = {
                             "bmap": bmap,
                             "mode_vn": mode_vn,
+                            "mods": mods,
```

Comment:
mods can be assigned as None here so the type of `LastNp.mods` should be `Mods | None`

if u use vs code, you can set the `type checking mode` setting to `"basic"` to get type checking in your editor

![image](https://i.cmyui.xyz/-imNCLx25Av36t2DNXE.png)

![image](https://i.cmyui.xyz/U7yk_pQ_c7x93J6gDw.png)

Reply:
fixed